drivers/hud_path_report/app/models/hud_path_report/generators/fy2020/base.rb

```diff
@@ -0,0 +1,176 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# require 'get_process_mem'
+module HudPathReport::Generators::Fy2020
+  class Base < ::HudReports::QuestionBase
+    include ArelHelper
+    include CommonQueries
+    include HudReports::Incomes
+
+    def initialize(generator = nil, report = nil, options: {})
+      super
+      @filter = HudPathReport::Filters::PathFilter.new.set_from_params(options)
+    end
+
+    private def universe
+      add_clients unless clients_populated?
+      @universe ||= @report.universe(self.class.question_number)
+    end
+
+    private def add_clients
+      client_scope.find_in_batches(batch_size: 100) do |batch|
+        pending_associations = {}
+        batch.each do |client|
+          enrollment = last_enrollment(client)
+          next unless enrollment.present?
+
+          source_client = enrollment.client
+
+          max_disability_date = enrollment.disabilities.select { |d| d.InformationDate <= @report.end_date }.
+            map(&:InformationDate).max
+          disabilities_latest = enrollment.disabilities.select { |d| d.InformationDate == max_disability_date }
```

Comment:
Is there a reason not to use `max_by`? https://apidock.com/ruby/Enumerable/max_by

Reply:
Never mind, this is getting all records from the date.